### ISSUE TITLE
Multilingual facets

### ISSUE THREAD

**Issue by ayalon** [algolia/instantsearch.js#3106]
Hi

We try to index multilingual facets. 

Example in the database:

facet_id | "Name DE" | "Name EN"
--------|-------------|--------------
1            | "Label DE"   | "Label EN"

In the search vue app we need mixed languages in the results. This means, for DE results we index "Label DE" and for EN results we index "Label EN".

The main issue is, that the same facet will appear as "Label DE" and "Label EN". But in fact, this is only one facet with the facet_id.

How can we aggregate multilingual facets and reducing them to an ID?

It feels, like Algolia has a very limited multilingual support.

**Comment by Haroenv** [comment]
Would the search be on both languages at the same time, or would it rather be that you can search in only one language at the same time? 

Is the situation that there's only one canonical dataset with translations, or should the filtering take the language in account? 

Could you give some more info on what exactly this means for the search?

**Comment by ayalon** [comment]
Thanks for your response!

We have a mixed dataset containing datasets in three different languages. When the user searches, he expects datasets from all languages at the same time. This sounds a bit uncommon, but is absolutely crucial in our use case.

Our data-provider has very structured datasets. Meaning all datasets have exactly the same structure and facets. The facet IDs are shared amongst all datasets. But the labels of the facets are localized in the according language.

We found no way to handle this use case. For example we would prefer the facet IDs in the url and not the localized facet names.

Do you have an idea how to solve that?

**Comment by Haroenv** [comment]
Just to confirm, since you opened this issue in InstantSearch.js. Are you using InstantSearch.js or Vue InstantSearch?

**Comment by ayalon** [comment]
You are right. We are using Vue instantsearch. Do I have to repost thr wholr issue?

**Comment by Haroenv** [comment]
No that's fine. What you want to do is the following for the current version of Vue InstantSearch:

1. use `ais-refinement-list`
2. fill in the default slot:

```vue
<template>
  <ais-refinement-list attribute-name="something">
    <template slot="default" slot-scope="{ value, active, count}">
      <div>
        <span>{{translationTable[value]}}</span>
        <span>{{count}}</span>
      </div>
    </template>
  </ais-refinement-list>
</template>

<script>
export default {
  data() {
    return {
      translationTable: {
        '1': {
          de: 'label fur eins',
          en: 'label for one',
        },
        '2': {
          de: 'label fur zwei',
          en: 'label for two',
        },
      },
    };
  },
};
</script>
```

In the next version (that will soon have an alpha) we have a better solution for this using `transformData`

Feel free to reopen if this didn't answer your question

**Comment by ayalon** [comment]
Correct me, if I am wrong:
This means, we have to index the "facet id" only and there will be no match on the facet label and also no query rules are possible, if a facet label is matched by the search term..

Coming from Solr, I wanted to index a facet as id / label pair, but it seems, that this is not possible?

In order to have the benefit of facet label matches and query rules, we tried to replace the text label with ids in the url and in the frontend with a larger hash table. But it feels so wrong to replace text-strings with id just to achieve proper multilingual support.

**Comment by Haroenv** [comment]
I understand the problem you are having, but as you already saw, in Algolia a facet is a single unique data point in the index. A label would be left up to the view layer, which in case of translation needs to be implemented with a hash like that. 

I added the feedback label to make it clear that this is a limitation, but I don't see any different solution than the one I already suggested. I hope you understand that :)
